src/parallelization/hpxsimulator.cpp

```diff
@@ -6,12 +6,16 @@
 namespace LibGeoDecomp {
 namespace HpxSimulatorHelpers {
 
+hpx::lcos::local::promise<void> initPromise;
```

Comment:
On 02/12/2016 02:25 PM, Andreas Schäfer wrote:
> In src/parallelization/hpxsimulator.cpp
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/3#discussion_r52740010>:
> 
>> @@ -6,12 +6,16 @@
>>  namespace LibGeoDecomp {
>>  namespace HpxSimulatorHelpers {
>>  
>> +hpx::lcos::local::promise<void> initPromise;
> 
> The basename is expected to be unique to the simulation, yes.

That's a problem then. I guess we need a mutex there to protect those
maps ... otherwise you easily run into race conditions there...
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/3/files#r52740010>.
> 


-- 
Thomas Heller
Friedrich-Alexander-Universität Erlangen-Nürnberg
Department Informatik - Lehrstuhl Rechnerarchitektur
Martensstr. 3
91058 Erlangen
Tel.: 09131/85-27018
Fax:  09131/85-27912
Email: thomas.heller@cs.fau.de